identity-core/src/crypto/signature/ed25519.rs

```diff
@@ -38,42 +46,49 @@ where
 {
   type Public = T;
 
-  fn verify(message: &[u8], signature: &[u8], key: &Self::Public) -> Result<()> {
-    let key: ed25519::PublicKey = parse_public(key.as_ref())?;
-    let sig: ed25519::Signature = parse_signature(signature)?;
+  type Error = VerificationError;
+
+  fn verify(message: &[u8], signature: &[u8], key: &Self::Public) -> std::result::Result<(), Self::Error> {
+    let key: ed25519::PublicKey = parse_public(key.as_ref()).map_err(Self::Error::from)?;
+    let sig: ed25519::Signature = parse_signature(signature).map_err(|err| VerificationProcessingError::from(err.0))?;
 
     if key.verify(&sig, message) {
       Ok(())
     } else {
-      Err(Error::InvalidProofValue("ed25519"))
+      Err(ProofValueError("ed25519").into())
     }
   }
 }
 
-fn parse_public(slice: &[u8]) -> Result<ed25519::PublicKey> {
+fn parse_public(slice: &[u8]) -> std::result::Result<ed25519::PublicKey, KeyError> {
   let bytes: [u8; PUBLIC_KEY_LENGTH] = slice
     .try_into()
-    .map_err(|_| Error::InvalidKeyLength(slice.len(), PUBLIC_KEY_LENGTH))?;
-
-  ed25519::PublicKey::try_from_bytes(bytes).map_err(Into::into)
+    .map_err(|_| KeyError("could not create a public key from the supplied bytes: incorrect length"))?;
+  ed25519::PublicKey::try_from_bytes(bytes).map_err(|_| KeyError("could not parse public key from the supplied bytes"))
 }
 
-fn parse_secret(slice: &[u8]) -> Result<ed25519::SecretKey> {
+fn parse_secret(slice: &[u8]) -> std::result::Result<ed25519::SecretKey, KeyError> {
```

Comment:
Nit: The `std::result::` is technically unnecessary. Is this for clarity? Could also be imported as `StdResult`, but it's fine either way.

Reply:
Thanks for pointing that out. I temporarily introduced it while the old crate level `Result` type was also in scope, but forgot to clean this up after deleting the old Result type.